conans/client/conf/config_installer.py

```diff
@@ -80,8 +81,9 @@ def _filecopy(src, filename, dst):
     src = os.path.join(src, filename)
     dst = os.path.join(dst, filename)
     if os.path.exists(dst):
+        os.chmod(dst, stat.S_IWRITE)  # make file writable
         os.remove(dst)
-    shutil.copy(src, dst)
+    shutil.copyfile(src, dst)
```

Comment:
I am concerned that this might break for other users. We had the case of @fulara that used different users (the reason the above ``os.remove()`` was introduced). ``copyfile`` copy with permissions, something that was avoided.

Isn't it possible to achieve the same without changing this line?

Reply:
Actually in my origianl request:
https://github.com/conan-io/conan/issues/6556

I explained how in my opinion the copyfile would resolve my issue, 
it is because the copyfile unlike copy does not change the owner of the file if invoked across different users. 

In this case though the file is removed if it exists  so this should be fine.

Reply:
> I am concerned that this might break for other users. We had the case of @fulara that used different users (the reason the above `os.remove()` was introduced). `copyfile` copy with permissions, something that was avoided.
> 
> Isn't it possible to achieve the same without changing this line?

`shutil.copyfile` copies without permissions :-) 

`shutil`: 

`copy() copies the file data and the file’s permission mode (see os.chmod()). Other metadata, like the file’s creation and modification times, is not preserved. To preserve all file metadata from the original, use`

`copyfile(): Copy the contents (no metadata) of the file [...]`

Source: https://docs.python.org/3.7/library/shutil.html